x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/ClusterStateRoleMapper.java

```diff
@@ -81,7 +82,22 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
     }
 
-    private Set<ExpressionRoleMapping> getMappings() {
+    public boolean hasMapping(String name) {
+        return getMappings().stream().map(ExpressionRoleMapping::getName).anyMatch(name::equals);
+    }
+
+    public Set<ExpressionRoleMapping> getMappings(@Nullable Set<String> names) {
+        if (enabled == false) {
+            return Set.of();
+        }
+        final Set<ExpressionRoleMapping> mappings = getMappings();
+        if (names == null || names.isEmpty()) {
+            return mappings;
+        }
+        return mappings.stream().filter(it -> names.contains(it.getName())).collect(Collectors.toSet());
+    }
+
+    public Set<ExpressionRoleMapping> getMappings() {
```

Comment:
nit: I think it would be cleaner to turn this around, so that the `getMappings()` method calls the `getMappings(@Nullable Set<String> names)` with `null` instead.

Reply:
👍